datachannel.go

```diff
@@ -172,6 +173,11 @@ func (d *DataChannel) open(sctpTransport *SCTPTransport) error {
 	d.mu.Unlock()
 
 	d.handleOpen(dc)
+
+	if d.api.settingEngine.detach.DataChannels && !d.detached {
```

Comment:
In the current test code, `Detach` is called from the another goroutine. The test in now failing. (https://travis-ci.org/pion/webrtc/builds/623092448#L452)
https://github.com/pion/webrtc/blob/aa582df55e7470c4f948598a445492a493d61d6a/datachannel_go_test.go#L352-L390

Also, read of `d.detached` is causing data race on CI.
This should be `d.mu.Lock()`-ed, if `Detach` is allowed to be called from goroutine.

Reply:
Thanks for the feedback @at-wat . I think @Sean-Der already made some modifications to my changes. If this is still an issue please let me know. I was also wondering if this additional log message is needed in `datachannel_js.go`.

Reply:
Thanks for the feedback, @at-wat. Is this still an issue after the changes @Sean-Der made? I will look into resolving it if this is still a problem. I am also wondering if a similar message is required in `datachannel_js.go`?

Reply:
Yes, the same error occurs on the master branch. (above link to Travis-CI is on the master branch.)
But, I'm not very sure that `Detach` is allowed to be called outside `OnOpen` handler by design.
If it is not allowed, the test code should be fixed; otherwise, the implementation should be fixed.

Reply:
Yea I messed that up :( fixing now! Will send a PR to you both.

@langep I am going to add you to the Pion organization so you can approve the PR. That also means when other people approve your PRs you can now merge them, excited to have you involved!